api/src/decorators/set-user.interceptor.ts

```diff
@@ -0,0 +1,21 @@
+import { CallHandler, ExecutionContext, NestInterceptor } from '@nestjs/common';
+import { Observable } from 'rxjs';
+
+export class SetUserInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    const request: any = context.switchToHttp().getRequest();
+    if (!request.user) return next.handle();
+
+    switch (request.method) {
+      case 'PUT':
+      case 'PATCH':
+        request.body.updatedById = request.user.id;
```

Comment:
updatedById should also be set on create. The goal is that we can make them not nullable